### ISSUE TITLE
Will not load since latest master updates

### ISSUE THREAD

**Issue by radicaled06** [mikepauer/Carbonite#38]
Since the master updates roughly 3 hours ago, carbonite is completely broken and will not load at all. #

**Error Code:**
Date: 2017-08-18 21:21:54
ID: 1
Error occured in: Global
Count: 1
Message: ..\AddOns\Carbonite\Carbonite.lua line 878:
   Usage: RegisterEvent("eventname", "methodname"): 'methodname' - method 'CHANNEL_ROSTER_UPDATE' not found on self.
Debug:
   [C]: ?
   [C]: ?
   ...ags\libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:138: RegisterEvent()
   Carbonite\Carbonite.lua:878: InitEvents()
   Carbonite\Carbonite.lua:778: SetupEverything()
   Carbonite\Carbonite.lua:1246: NXOnUpdate()
   [string "*:OnUpdate"]:1:
      [string "*:OnUpdate"]:1

**Comment by radicaled06** [comment]
Update:

Through some trial and error it does seem to be a conflict with my WeakAuras addon. I'm not sure how to fix this?

**Comment by funkydude** [comment]
https://github.com/mikepauer/Carbonite/blob/d91f6bf9366d06ea538287278de52bd89fd12783/Carbonite.lua#L845

It's not a conflict with anything, the error is clear as day. Addon is trying to register the event `CHANNEL_ROSTER_UPDATE` with the function `Com.OnChannel_roster_update` which doesn't exist, so naturally, AceEvent tries to fall back to finding a function called `CHANNEL_ROSTER_UPDATE` which also doesn't exist.

TLDR it's a valid error the authors need to resolve instead of [upgrading](https://github.com/mikepauer/Carbonite/commit/8cd28a878b1e96ccca009c236481351a39e3f871) and then [downgrading](https://github.com/mikepauer/Carbonite/commit/89891403841d066ff5858a4414f70116904aeba2) their ace libraries over and over again.

**Comment by IrcDirk** [comment]
Shit they did downgrade again????

**Comment by IrcDirk** [comment]
My fault. Fixed.

**Comment by funkydude** [comment]
You re-upgraded the libraries, you didn't fix the source of the error which I highlighted. If you don't require the event anymore, that line can simply be removed.

**Comment by funkydude** [comment]
@IrcDirk You re-upgraded the libraries, you didn't fix the source of the error which I highlighted. If you don't require the event anymore, that line can simply be removed.

**Comment by mikepauer** [comment]
It's fixed now, surprised an entirely missing function took 4-5 years to finally produce an error in the code

As for ace libraries.. we shouldn't ever be using alpha code in a public/stable release, no matter what other addons decide to do so.

**Comment by radicaled06** [comment]
Well that's strange, after I first posted I did some trial and error and disabled my other addons one-by-one until I found the issue. After disabling WA it worked perfectly fine. 

Thanks for the fix, though!

**Comment by funkydude** [comment]
@mikepauer It's actually really simple. AceEvent embeds the function "RegisterEvent" in your addon. However, for whatever unknown reason, you guys create your own version of "RegisterEvent" here:

https://github.com/mikepauer/Carbonite/blob/590a02a96eeedbdb0a1b4deb470b6a639390ce62/NxUI.lua#L2766

This overwrites the AceEvent version, and prevents the error from showing up.

The reason addons with newer versions of AceEvent (such as WeakAuras) revealed this is as follows: When the newer AceEvent loads, it loops through all the addons that embed it, and overwrites the "RegisterEvent" function with the AceEvent one.

Before AceEvent update:
You embed AceEvent.
You overwrite AceEvent's "RegisterEvent" so the error never shows.

After AceEvent update:
You embed AceEvent.
You overwrite AceEvent's "RegisterEvent"
An upgrade happens, and overwrites your version of "RegisterEvent" back to AceEvent's version of it, allowing errors to show once again.

You should probably investigate why you have a function called RegisterEvent when you are embedding AceEvent.

**Comment by Thrumbar** [comment]
The use if RegisterEvent in said fashion goes back before Carbonite used any Ace libs and is exactly as used during MoP release.  Was not changed when the addon was moded with the release of WoD.

So either we could change to AceEvent or stay and drop AceEvent from libs..

**Comment by funkydude** [comment]
Either drop AceEvent and handle events yourself or remove your own RegisterEvent function.

Otherwise this issue will happen again next time there's an AceEvent update.

**Comment by mikepauer** [comment]
Converting to AceEvent.. originally I was going to drop it, but almost all of Carbonite is already using it. It's only Quests and Maps that were not... the other Modules I had already converted to using it when I split everything into modular.
